### PR TITLE
feat-restore-certs

### PR DESCRIPTION
**Pull request by pagottoo**
Download SSL/TLS files from S3 and apply on k8s cluster to avoid re-issuer certificates.

### CODE REVIEW

File: cmd/restoreSSL.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
```
Comment (converge) [inline cmd/restoreSSL.go:1]: it would be nice to remove left overs

File: cmd/restoreSSL.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("restoreSSL called")
+		ssl.RestoreSSL()
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(restoreSSLCmd)
+
+	// Here you will define your flags and configuration settings.
```
Comment (converge) [inline cmd/restoreSSL.go:33]: it would be nice to remove left overs

File: internal/aws/aws.go [hunk 3]
```diff
@@ -339,3 +341,106 @@ func DestroyBucketsInUse(destroyBuckets bool) {
...
+	downloader := s3manager.NewDownloader(GetAWSSession())
+
+	log.Println("Listing the objects in the bucket:")
+	listObjsResponse, err := s3Client.ListObjectsV2(&s3.ListObjectsV2Input{
+		Bucket: aws.String(bucket),
+		Prefix: aws.String(""),
+	})
+
+	if err != nil {
+		panic("Couldn't list bucket contents")
```
Comment (converge) [inline internal/aws/aws.go:424]: would be nice to log it

File: internal/ssl/ssl.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	"github.com/kubefirst/kubefirst/internal/aws"
+	"github.com/kubefirst/kubefirst/internal/k8s"
+	"github.com/kubefirst/kubefirst/pkg"
+	"github.com/spf13/viper"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var namespaces []string
```
Comment (converge) [inline internal/ssl/ssl.go:22]: it would be a good idea to remove globals, we can host it inside Configs

File: internal/aws/aws.go [hunk 5]
```diff
@@ -330,3 +332,107 @@ func DestroyBucketsInUse(destroyBuckets bool) {
...
+	} else {
+		log.Printf("[#99] Dry-run mode, bucket creation skipped:  %s", name)
+	}
+	viper.Set(fmt.Sprintf("bucket.%s.created", name), true)
+	viper.Set(fmt.Sprintf("bucket.%s.name", name), name)
+	viper.WriteConfig()
+}
+
+func UploadFile(bucket, key, fileName string) error {
+	// The session the S3 Uploader will use
```
Comment (converge) [inline internal/aws/aws.go:375]: *left over

File: internal/aws/aws.go [hunk 6]
```diff
@@ -330,3 +332,107 @@ func DestroyBucketsInUse(destroyBuckets bool) {
...
+	})
+	if err != nil {
+		return fmt.Errorf("failed to upload file, %v", err)
+	}
+	log.Printf("file uploaded to, %s\n", result.Location)
+	return nil
+}
+
+func DownloadBucket(bucket string, destFolder string) error {
+	// Create a file to write the S3 Object contents to.
```
Comment (converge) [inline internal/aws/aws.go:400]: *left over

File: internal/ssl/ssl.go [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+				return nil, err
+			}
+			files = append(files, fullFileName)
+		}
+	}
+
+	return files, nil
+}
+
+//func GetBackupCertificates(apiGroup string, apiVersion string, resourceTypes []string, namespace string) ([]string, error) {
```
Comment (converge) [inline internal/ssl/ssl.go:85]: *left over

File: internal/ssl/ssl.go (hunk 4 above)
Comment (pagottoo) [reply internal/ssl/ssl.go]: resolved using a func to return the list of namespaces

**Review by converge — APPROVED** [review]

**Comment by johndietz** [comment]
thanks for getting this in today guys

**Pull request merged**